staging/kos/cmd/controller-manager/main.go

```diff
@@ -55,21 +55,21 @@ func main() {
 		os.Exit(2)
 	}
 
-	kosInformersClientset, err := kosclientset.NewForConfig(kosClientCfg)
+	kosInformersClientset, err := kosclientset.NewForConfig(addUserAgent(kosClientCfg, "kos-controller-manager"))
 	if err != nil {
 		glog.Errorf("Failed to configure kos clientset for informers: %s.", err.Error())
 		os.Exit(3)
 	}
 
 	ctx := controllerContext{
-		k8sClientCfg:    k8sClientCfg,
-		kosClientCfg:    kosClientCfg,
 		options:         ctlrOpts,
 		sharedInformers: kosinformers.NewSharedInformerFactory(kosInformersClientset, 0),
 		stop:            stopOnSignals(),
 	}
 	for controller, startController := range managedControllers {
-		if err := startController(ctx); err != nil {
+		k8sCC := addUserAgent(k8sClientCfg, controller)
+		kosCC := addUserAgent(kosClientCfg, controller)
```

Comment:
How about we move the setting of the user agent within the start functions?

Reply:
That would be duplicating code.